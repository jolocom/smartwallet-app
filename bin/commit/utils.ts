import childProcess from 'child_process'

export const promisify =
  <T>(fn: (args: T) => void) =>
  (args: T): Promise<void> =>
    new Promise((res, rej) => {
      try {
        fn(args)
        res()
      } catch (e) {
        rej(e)
      }
    })

export const listStagedFiles = () =>
  childProcess.execSync(`git diff --cached --name-only --diff-filter=ACMR`)

export const stageModifiedFiles = (stagedFiles: string) =>
  childProcess.execSync(`echo "${stagedFiles}" | xargs git add`)

export const abortScript = (msg: string) => {
  console.log('')
  console.log('\x1b[0;33m%s\x1b[0m', msg)
  console.log('\x1b[0;31m%s\x1b[0m', 'Aborting')
  process.exit(1)
}

export const logStep = (msg: string) => {
  console.log('\x1b[0;34m%s\x1b[0m', `${msg}`)
}

export const spawnProcess = (
  onClose: (
    code: number | null,
    dataOutput: string,
    errorOutput: string,
  ) => void,
  cmd: string,
  args?: readonly string[],
  options?: childProcess.SpawnOptionsWithoutStdio,
) => {
  const spawnedProcess = childProcess.spawn(cmd, args, options)

  let dataOutput = ''
  let errorOutput = ''

  spawnedProcess.stdout.on('data', (data) => {
    dataOutput += data.toString() + '\n'
  })
  spawnedProcess.stderr.on('data', (error: Buffer) => {
    errorOutput += error.toString() + '\n'
  })

  spawnedProcess.stderr.pipe(process.stderr)
  spawnedProcess.stdout.pipe(process.stdout)

  spawnedProcess.on('close', (code) => onClose(code, dataOutput, errorOutput))

  return spawnedProcess
}
