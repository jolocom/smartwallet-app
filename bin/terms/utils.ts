import fetch from 'node-fetch';
import fs from 'fs';
import shell from 'shelljs'

const CURRENT_PATH = process.cwd();
const SECRETS_REPO = 'dev@hetz1.jolocom.io:poeditor-key'
const BASE_URL = 'https://api.poeditor.com/v2';
const API_KEY_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_TOKEN.txt`
const PROJECT_ID_LOCATION = `${CURRENT_PATH}/bin/poeditor-key/POEDITOR_SW2_PROJECT_ID.txt`

export enum Languages {
  en = 'en',
  de = 'de',
}

interface IResult  {
  [key: string]: {
    [key: string]: number
  }
}

export interface IResponse<T = IResult> {
  response: {
    status: string,
    code: string,
    message: string,
  },
  result: T
}

export const cloneSecrets = () => {
  return new Promise((res, rej) => {
    const keyExists = fs.existsSync(API_KEY_LOCATION);
    const idExists = fs.existsSync(PROJECT_ID_LOCATION);
    if(!keyExists || !idExists) {
      shell.cd(`${CURRENT_PATH}/bin`)
      shell.rm('-rf', `poeditor-key`)
      shell.exec(
        `git clone ${SECRETS_REPO} -b master`,
        {},
        (code, stdout, stderr) => {
          if (code != 0) return rej(new Error(stderr))
          return res(stdout)
        },
      )
    } else {
      res('Secrets are already present');
    }
  })
}

const readTextFile = (location: string) => {
  return new Promise<string>((res, rej) => {
    fs.readFile(location, 'utf8', (err, data) => {
      if (err) rej(err)
      else res(data.toString().trim())
    })
  })
}

const getPoeditorSecrets = async () => {
    const apiKey = await readTextFile(API_KEY_LOCATION)
    const projectId = await readTextFile(PROJECT_ID_LOCATION)
    return { apiKey, projectId }
}


export const sendPostRequest = async <T = IResponse, >(path: string, params: Record<string, string>) => {
  try {
    const { apiKey, projectId } = await getPoeditorSecrets();
    const requestParams = new URLSearchParams({
      api_token: apiKey,
      id: projectId,
      ...params
    })
    const res = await fetch(`${BASE_URL}${path}`, {
      method: 'POST',
      body: requestParams.toString(),
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    })
    return await res.json() as T;
  } catch(err) {
    console.error(err);
  }
}