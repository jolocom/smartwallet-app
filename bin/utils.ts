import fs from 'fs'
import { stringify as stringifyEnv, parse as parseEnv, Data } from 'envfile'
import shell from 'shelljs'

export const cloneRepo = async (remote: string, repoName: string) => {
  return new Promise<string>((res, rej) => {
    const dir = shell.tempdir()
    console.log(`${dir}/${repoName}`)
    shell.rm('-rf', `${dir}/${repoName}`)
    shell.exec(
      `git clone ${remote}:${repoName} -b master ${dir}/${repoName}`,
      {},
      (code, stdout, stderr) => {
        if (code != 0) return rej(new Error(stderr))
        return res(`${dir}/${repoName}`)
      },
    )
  })
}

export const catFile = async (path: string) => {
  return new Promise((res, rej) => {
    const dsn = shell.cat(path)
    if (dsn.stderr) rej(dsn.stderr)
    res(dsn.stdout)
  })
}

export const writeEnv = async (path: string, data: object) => {
  const env = stringifyEnv(data)

  return new Promise((res, rej) => {
    fs.writeFile(path, env, (err) => {
      if (err) rej(err)
      else res(undefined)
    })
  })
}

export const readEnv = async (path: string) => {
  return new Promise<Data>((res, rej) => {
    fs.readFile(path, 'utf8', (err, data) => {
      if (err) {
        console.log(
          '\x1b[4;31m%s\x1b[0m',
          'Error! Could not find the environmental config (.env).',
        )
      } else res(parseEnv(data))
    })
  })
}
