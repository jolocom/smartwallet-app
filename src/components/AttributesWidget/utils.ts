import { AttrsState } from '~/modules/attributes/types'
import { SelectableAttrI } from './index'

export const getMappedAttrubited = (attrs: AttrsState<string>) =>
  Object.keys(attrs).reduce((acc, val) => {
    acc[val] = attrs[val].map((attr) => ({
      val: attr,
      isSelected: false,
    }))
    return acc
  }, {} as AttrsState<SelectableAttrI>)
