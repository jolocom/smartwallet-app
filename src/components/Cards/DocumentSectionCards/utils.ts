export const splitFields = <T>(fields: Array<T>) => {
  const halfway = Math.ceil(fields.length / 2)
  const last = fields.splice(halfway, fields.length)
  const first = fields.splice(0, halfway)

  return [first, last]
}
