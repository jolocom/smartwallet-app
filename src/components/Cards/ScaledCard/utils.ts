import {
  LayoutChangeEvent,
  StyleSheet,
  TextStyle,
  ViewStyle,
} from 'react-native'
import { TSupportedComponentProps } from './types'

/**
 * @internal
 */
export const scaleStyleObject = <T extends TSupportedComponentProps['style']>(
  style: T,
  scaleBy: number,
) => {
  style = StyleSheet.flatten(style) as T

  return Object.entries(style ?? {}).reduce<ViewStyle | TextStyle>(
    (acc, [key, value]) => ({
      ...acc,
      [key]: Number.isInteger(value) ? value * scaleBy : value,
    }),
    {},
  )
}

/**
 * @internal
 */
export const handleContainerLayout = (e: LayoutChangeEvent) => {
  const { width, height } = e.nativeEvent.layout
  return {
    width,
    height,
  }
}
