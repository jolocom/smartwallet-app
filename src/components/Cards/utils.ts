export const getTrimmedHighlight = (highlight?: string, nrChar = 14) => {
  if (highlight) {
    return highlight?.length > nrChar
      ? highlight?.slice(0, nrChar) + '...'
      : highlight
  } else {
    return undefined
  }
}
