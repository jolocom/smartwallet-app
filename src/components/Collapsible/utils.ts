import { TTitle } from './types'

export const compare = (a: TTitle, b: TTitle): number => {
  if (a.startY < b.startY) {
    return -1
  }
  if (a.startY > b.startY) {
    return 1
  }
  return 0
}
