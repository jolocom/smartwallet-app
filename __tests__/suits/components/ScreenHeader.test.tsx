import React from 'react'
import ScreenHeader from '~/components/ScreenHeader'
import renderer from 'react-test-renderer'
import { renderWithSafeArea } from '../../utils/renderWithSafeArea'

describe('ScreenHeader', () => {
  const defaultProps = {
    title: 'title text',
    subtitle: 'subtitle text',
  }

  it('renders correctly', () => {
    const { toJSON } = renderer.create(<ScreenHeader />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render title correctly', () => {
    const { getByTestId } = renderWithSafeArea(
      <ScreenHeader {...defaultProps} />,
    )
    expect(getByTestId('title').props.children).toBe(defaultProps.title)
  })

  it('should render subtitle correctly', () => {
    const { getByTestId } = renderWithSafeArea(
      <ScreenHeader {...defaultProps} />,
    )
    expect(getByTestId('subtitle').props.children).toBe(defaultProps.subtitle)
  })
})
