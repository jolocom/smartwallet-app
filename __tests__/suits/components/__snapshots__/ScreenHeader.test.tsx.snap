// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenHeader renders correctly 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "paddingTop": 5,
          "textAlign": "center",
        },
        Object {
          "color": "rgba(255,255,255,0.85)",
          "fontFamily": "TTCommons-Regular",
          "fontSize": 28,
          "letterSpacing": 0,
          "lineHeight": 32,
        },
        undefined,
      ]
    }
    testID="title"
  >
    title text
  </Text>
  <Text
    style={
      Array [
        Object {
          "paddingTop": 5,
          "textAlign": "center",
        },
        Object {
          "color": "rgba(255, 255, 255, 0.7)",
          "fontFamily": "TTCommons-Regular",
          "fontSize": 20,
          "letterSpacing": 0.14,
          "lineHeight": 22,
        },
        Object {
          "opacity": 0.8,
          "paddingTop": 16,
        },
      ]
    }
    testID="subtitle"
  >
    subtitle text
  </Text>
</View>
`;
